plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObjectPdo.class.php

```diff
@@ -1063,6 +1063,49 @@ private function getActRights()
     return $actRights;
   }
 
+  /**
+   * Get an array outlining which top level description the current description is part of.
+   *
+   * @param int $collectionRootId  The id for the top level description in this hierarchy.
+   * @return array  An array containing the id and various i18n titles of the top level description.
+   *                If this description is already a TLD or another issue occurs, return null.
+   */
+  public function getPartOf($collectionRootId)
+  {
+    $collectionRootId = $this->getCollectionRootId();
+
+    if ($collectionRootId && $collectionRootId != $this->id)
+    {
+      $rootSlug = QubitPdo::fetchColumn('SELECT slug FROM slug WHERE object_id=?', array($collectionRootId));
+      if (!$rootSlug)
+      {
+        throw new sfException("No slug found for information object $collectionRootId");
```

Comment:
Throwing an exception in here will avoid the entire document to be added to the ES index. I think it's better to have it without part of the `parOf` data than not having the document in the index.

Reply:
The first condition in the if is from if a result from a db query indicating an info object exists. If one exists without a slug and isn't ROOT_ID, I think that's data corruption? I don't think it's wise to continue the code path here.

Reply:
I agree this is a data corruption issue on the ancestor, but I don't think it's enough to remove all the descendants from the index. The search populate task will show an error at the end about the ancestor issue but, if this is called from the GUI, and somehow the ancestor lost its slug, you may end up loosing a lot of records without noticing.